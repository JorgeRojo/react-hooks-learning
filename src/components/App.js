import React from "react";
import { Provider } from "react-redux";
import store from "../store";

import CounterHeader from "./Counter/CounterHeader";
import CounterForm from "./Counter/CounterForm";

function App() {
  return (
    <Provider store={store}>
      <CounterHeader />
      <CounterForm />
    </Provider>
  );
}

export default App;
