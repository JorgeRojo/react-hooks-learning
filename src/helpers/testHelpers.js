import React from 'react';

export const mockComponent = (name, props) => (
  <code name={name}>{JSON.stringify(props)}</code>
);
