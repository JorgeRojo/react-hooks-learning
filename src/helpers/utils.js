import _get from 'lodash/get';

export const getNumberFromEvent = event => {
  const value = _get(event, 'target.value', '');
  if (value === '') {
    return '';
  }
  const number = parseInt(value, 10);
  return !isNaN(number) ? number : 0;
};
