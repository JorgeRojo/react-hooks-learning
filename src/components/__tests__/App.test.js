import React from 'react';
import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';

import App from '../App';
import { mockComponent } from '../../helpers/testHelpers';

jest.mock('../pages/Counter/CounterHeader', () => props =>
  mockComponent('CounterHeader', props)
);
jest.mock('../pages/Counter/CounterForm/CounterForm', () => props =>
  mockComponent('CounterForm', props)
);

describe('App Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the component', () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });
});
