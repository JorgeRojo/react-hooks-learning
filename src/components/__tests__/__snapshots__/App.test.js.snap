// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders the component 1`] = `
<div>
  <nav
    style="position: absolute;"
  >
    <ul>
      <li>
        <a
          href="/"
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/counter"
        >
          Counter
        </a>
      </li>
      <li>
        <a
          href="/posts"
        >
          Posts
        </a>
      </li>
    </ul>
  </nav>
  <div
    class="container text-center mt-4"
  >
    <h1
      class="display-4 mb-4"
    >
      Home
    </h1>
    <p>
      Testing react hooks
    </p>
  </div>
</div>
`;
